### PR TITLE
[Dataprotection] Onboard ResourceGuard and AzMon alerts

### PR DESCRIPTION
**Pull request by akshayneema**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [ ] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [ ] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
The precondition is to put your code inside this repository and upgrade the version in the pull request but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
Dataprotection

File: src/dataprotection/azext_dataprotection/generated/_params.py [hunk 1]
```diff
@@ -221,3 +229,26 @@ def load_arguments(self, _):
...
+    with self.argument_context('dataprotection resource-guard create') as c:
+        c.argument('resource_group_name', resource_group_name_type)
+        c.argument('resource_guards_name', options_list=['--resource-guard-name', '--name', '-n'], type=str, help='The '
+                   'name of ResourceGuard')
+        c.argument('e_tag', type=str, help='Optional ETag.')
+        c.argument('location', arg_type=get_location_type(self.cli_ctx), required=False,
+                   validator=get_default_location_from_resource_group)
+        c.argument('tags', tags_type)
+        c.argument('type_', options_list=['--type'], type=str, help='The identityType which can be either '
+                   'SystemAssigned or None', arg_group='Identity')
```
Comment (kairu-ms) [inline src/dataprotection/azext_dataprotection/generated/_params.py:247]: You should use `get_enum_type` for this argument.

File: src/dataprotection/azext_dataprotection/generated/_params.py [hunk 2]
```diff
@@ -46,11 +46,19 @@ def load_arguments(self, _):
         c.argument('type_', options_list=['--type'], type=str, help='The identityType which can be either '
                    'SystemAssigned or None', arg_group='Identity')
```
Comment (kairu-ms) [inline src/dataprotection/azext_dataprotection/generated/_params.py:47]: Same concern about it.

**Review by kairu-ms — APPROVED** [review]

**Review by kairu-ms — COMMENTED** [review]
Please update the version of this extension.

**Pull request merged**